src/modm/platform/adc/stm32/adc_impl.hpp.in

```diff
@@ -97,6 +97,12 @@ modm::platform::Adc{{ id }}::setChannel(const Channel channel,
 	ADC{{ per }}->SQR3 = uint32_t(channel) & 0x1f;
 
 	setSampleTime(channel, sampleTime);
+%% if temperature_available
+	if(channel == Channel::TemperatureSensor || channel==Channel::InternalReference)
+	{
+		ADC->CCR |= ADC_CCR_TSVREFE;
+	}
+%%endif
```

Comment:
```suggestion
%% if temperature_available
	if(channel == Channel::TemperatureSensor || channel == Channel::InternalReference) {
		enableTemperatureRefVMeasurement();
	}
%% endif
```

You have probably missed [enableTemperatureRefVMeasurement()](https://github.com/modm-io/modm/blob/04ed0a52e03964811eb9764306bb1b61de5b5f60/src/modm/platform/adc/stm32/adc_impl.hpp.in#L47).

On F1 devices there is no `CCR` register. `enableTemperatureRefVMeasurement()` will always do the right thing.